packages/algorand-js/src/runtime/account.ts

```diff
@@ -101,4 +102,45 @@ export class StoreAccountImpl implements StoreAccount {
       appId: appId
     });
   }
+
+  createApp (params: SSCDeploymentFlags): CreatedApps {
+    if (this.createdApps.length === 10) {
+      throw new Error('Maximum created applications for an account is 10');
+    }
+
+    const appId = Math.floor(1000 + Math.random() * 9000);
+    const appParams: SSCParams = {
+      'approval-program': '',
+      'clear-state-program': '',
+      creator: params.sender.addr,
+      'global-state': [],
+      'global-state-schema': { 'num-byte-slice': params.globalBytes, 'num-uint': params.globalInts },
+      'local-state-schema': { 'num-byte-slice': params.localBytes, 'num-uint': params.localInts }
+    };
+    // create new app in sender's account
+    const newApp: CreatedApps = { id: appId, params: appParams };
+    this.createdApps.push(newApp); // push newly created app
+
+    console.log('Created new app with id:', appId);
+    return newApp;
+  }
+
+  // opt in to application
+  optInToApp (appId: number, appParams: SSCParams): void {
+    const localState = this.appsLocalState.find(cfg => cfg.id === appId);
+    if (localState) {
+      console.warn(`app ${appId} already opted in to ${this.address}`);
+    } else {
+      if (this.appsLocalState.length === 10) {
+        throw new Error('Maximum Opt In applications per account is 10');
+      }
+
+      const localParams: AppLocalState = {
+        id: appId,
+        "key-value": [],
+        schema: appParams["local-state-schema"]
+      };
+      this.appsLocalState.push(localParams); // push
```

Comment:
> should we store in a map?
> 
> ```
> appLocalState: Map<app_id, AppLocalState>
> ```

I have a suggestion, extending to this comment, i think we should change the structure of Store, instead of storing them as in `algosdk` we can have them according to our requirements because we won't be interacting with algorand.

- For ex: In assets, we can directly use Map structure directly, instead of having, assets: {asset-id, assetbalance, ....}
we can directly store assets: {asset-id -> data}, 
this way we won't have to copy the data,(updating will be easier and faster too)
- in some places we are using traversing whole objects to find the value, it will be faster to use map instead.
what do you think ?

Reply:
> in some places we are traversing whole objects to find the value, it will be faster to use map instead.
what do you think ?
In `algorand` we only have max 10 applications per account. So using a map in that case seemed excessive.

I am also thinking to use maps instead. This way, i think we don't need to limit the app in account to 10. It will be more easier for end user also (getting data directly)

Reply:
yes, but each account may have long list of `appLocal-state`. as we will be in test environment multiple queries for the same application may occur. in terms of implementation and understanding complexity , it is easy to implement and understand so i don't think it will be a overkill.

Reply:
> yes, but each account may have long list of appLocal-state.

No, each account in `algorand` has also maximum 10 applications. So if we are limiting to 10, then simple search should be sufficient enough. 
But i don't think we should limit to 10 here in tests. Same for assets.
![image](https://user-images.githubusercontent.com/33264364/103091388-02807880-461a-11eb-912a-fbacce7caa1c.png)

Reply:
i meant, what if local state has storage variable `counter`, `counter1`, ........
all these variables have some value.  
For ex: 
```
export LOCAL_BYTESLICES=0
export LOCAL_INTS=10
```
state storage.
this means LOCAL_INTS has 10 int state storage for this application.

we can search directly if you want.

Reply:
> this means LOCAL_INTS has 10 int state storage for this application.

Yes, in local schema we can have 16 max key-value pairs. These are also stored as array.
> we can search directly if you want.

I think using map on the top level will be better with (maybe) increasing the limit.

Reply:
Update: Adding to TODO, and will do this in a new tasks. Some other structures needs to be updated as well.